src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/Program.cs

```diff
@@ -129,26 +128,70 @@ private static void Execute(string[] args, bool isNoBuild, ILogger logger)
+                configuration,
+                frameworkToUse,
+                server);
+
+            var exitCode = command
+                .OnErrorLine(e => logger.LogMessage(e, LogMessageLevel.Error))
+                .OnOutputLine(e => logger.LogMessage(e, LogMessageLevel.Information))
+                .Execute()
+                .ExitCode;
+            return exitCode;
+        }
+
+        // Creates a command to execute dotnet-aspnet-codegenerator-design
+        private static Command CreateDipatchCommand(
+            IProjectContext context,
+            string[] args,
+            string buildBasePath,
+            string configuration,
+            NuGetFramework frameworkToUse,
+            ScaffoldingServer server)
+        {
             var projectDirectory = Directory.GetParent(context.ProjectFullPath).FullName;
+
+            // Command Resolution Args
+            // To invoke dotnet-aspnet-codegenerator-design with the user project's dependency graph,
+            // we need to pass in the runtime config and deps file to dotnet for netcore app projects.
+            // For projects that target net4x, since the dotnet-aspnet-codegenerator-design.exe is in the project's bin folder
+            // and `dotnet build` generates a binding redirect file for it, we can directly invoke the exe from output location.
+
+            var targetDir = Path.GetDirectoryName(context.AssemblyFullPath);
+            var runtimeConfigPath = Path.Combine(targetDir, context.RuntimeConfig);
+            var depsFile = Path.Combine(targetDir, context.DepsFile);
+            var nugetPackageRoot = context.PackagesDirectory;
+
+            string dotnetCodeGenInsideManPath = context.CompilationAssemblies
+                .Where(c => Path.GetFileNameWithoutExtension(c.Name)
+                            .Equals(DESIGN_TOOL_NAME, StringComparison.OrdinalIgnoreCase))
+                .Select(reference => reference.ResolvedPath)
+                .FirstOrDefault();
+
+            if (string.IsNullOrEmpty(dotnetCodeGenInsideManPath))
+            {
+                throw new InvalidOperationException("Please add Microsoft.VisualStudio.Web.CodeGeneration.Design package to the project");
+            }
+
             var dependencyArgs = ToolCommandLineHelper.GetProjectDependencyCommandArgs(
                      args,
                      frameworkToUse.GetShortFolderName(),
                      server.Port.ToString());
 
-            var exitCode = DotnetToolDispatcher.CreateDispatchCommand(
-                    dependencyArgs,
-                    frameworkToUse,
-                    configuration,
-                    null,
-                    buildBasePath,
-                    projectDirectory,
-                    DESIGN_TOOL_NAME)
-                .ForwardStdErr()
-                .ForwardStdOut()
-                .Execute()
-                .ExitCode;
-            return exitCode;
+            return DotnetToolDispatcher.CreateDispatchCommand(
```

Comment:
I think you are basically the only one using this tool now. Can we just move that one class into your project and eliminate the sources package? (Out of scope for this PR, but let's open an issue for that so we can clean it up when we finally merge feature/msbuild to dev).